### PR TITLE
Adds ts_web_test_suite using rules_webtesting web_test_suite & browsers

### PR DESCRIPTION
**Pull request by gregmagolan**
Even better than #198.

Adds `ts_web_test_suite` with support for Firefox headless on top existing Chrome headless support.

```
ts_web_test_suite(
    name = "testing",
    deps = [
      ":tests",
    ],
    browsers = [
      "@io_bazel_rules_webtesting//browsers:chromium-local",
      "@io_bazel_rules_webtesting//browsers:firefox-local",
    ],
)
```

Leverages rules_webtesting `web_test_suite` rule & browser repositories. Depends on https://github.com/bazelbuild/rules_webtesting/pull/244, which updates the mac Firefox version to 58 to support headless Firefox on mac.

Implementation of `ts_web_test_suite` based on `wrap_web_test_suite` macro in rules_webtesting.

Also turns on node `preserve-symlinks` for repository.

### CODE REVIEW

**Comment by gregmagolan** [comment]
DO NOT MERGE.

Waiting on https://github.com/bazelbuild/rules_webtesting/pull/244

**Comment by gregmagolan** [comment]
Note: Some more testing required including that firefox & chrome windows show up with `bazel run` and that ibazel `ibazel_notify_changes` still works.

**Comment by gregmagolan** [comment]
https://github.com/bazelbuild/rules_webtesting/pull/244 landed and http_archive updated accordingly

File: defs.bzl [hunk 1]
```diff
@@ -20,15 +20,15 @@ load("//internal:ts_repositories.bzl", _ts_setup_workspace = "ts_setup_workspace
...
-load("//internal/karma:ts_web_test.bzl", _ts_web_test = "ts_web_test_macro")
+load("//internal/karma:ts_web_test_suite.bzl", _ts_web_test_suite = "ts_web_test_suite")
 load("//internal/protobufjs:ts_proto_library.bzl", _ts_proto_library = "ts_proto_library")
 
 ts_setup_workspace = _ts_setup_workspace
 ts_library = _ts_library
 ts_config = _ts_config
 ts_devserver = _ts_devserver
-# TODO(alexeagle): make ts_web_test work in google3
-ts_web_test = _ts_web_test
```
Comment (alexeagle) [inline defs.bzl]: Can we keep the name the same? Why change it to `_suite`?

Comment (gregmagolan) [reply defs.bzl]: Decided to go with ts_web_test_suite which is consistent with rules_webtesting and google3

File: .bazelci/presubmit.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ platforms:
     build_targets:
     - "..."
     test_flags:
-    - "--test_tag_filters=-browser:Chrome"
+    - "--test_tag_filters=-browser:chromium-local,-browser:firefox-local"
```
Comment (alexeagle) [inline .bazelci/presubmit.yml]: add a comment/TODO about why they are excluded

File: WORKSPACE [hunk 3]
```diff
@@ -55,12 +68,30 @@ go_rules_dependencies()
...
+    chromium = True,
+    firefox = True,
+)
+
+load("@build_bazel_rules_typescript//:defs.bzl", "ts_setup_workspace")
+
+ts_setup_workspace()
+
 #############################################
 # Dependencies for generating documentation #
```
Comment (alexeagle) [inline WORKSPACE]: please make sure `yarn skydoc` continues to work

File: WORKSPACE [hunk 4]
```diff
@@ -74,12 +105,6 @@ http_archive(
 load("@io_bazel_rules_sass//sass:sass.bzl", "sass_repositories")
 sass_repositories()
 
-http_archive(
```
[the remark this answers is not in the document]
Comment (alexeagle) [reply WORKSPACE]: happy to help diagnose that

File: defs.bzl [hunk 5]
```diff
@@ -20,15 +20,15 @@ load("//internal:ts_repositories.bzl", _ts_setup_workspace = "ts_setup_workspace
...
+# TODO(alexeagle): make ts_web_test_suite work in google3
+ts_web_test_suite = _ts_web_test_suite
```
Comment (alexeagle) [inline defs.bzl]: let's keep `ts_web_test` here as well so we don't break Angular

File: WORKSPACE (hunk 3 above)
Comment (gregmagolan) [reply WORKSPACE]: Still good

File: defs.bzl (hunk 5 above)
Comment (gregmagolan) [reply defs.bzl]: Done

File: .bazelci/presubmit.yml (hunk 2 above)
Comment (gregmagolan) [reply .bazelci/presubmit.yml]: :thumb:

**Comment by gregmagolan** [comment]
`ibazel run :target` tested and working as expected

File: WORKSPACE (hunk 4 above)
[the remark this answers is not in the document]
Comment (gregmagolan) [reply WORKSPACE]: what is @bazel_skylib used for? I don't see any uses and `yarn skydoc` still works.

**Comment by alexeagle** [comment]
It provides the `skylark_library` function, but yeah that never worked, I
guess this is dead code. As long as you can run the docgen, that satisfies
the requirements...

**Comment by gregmagolan** [comment]
Closed with d25f3ce7438d8fb3497115f04a029232c8ef2861

**Pull request closed without merging**